cortico.js

```diff
@@ -995,10 +1060,90 @@ async function checkAllEligibility() {
       total: length,
       error
     })
-    
+
   }
-  
-  
+
+
+}
+
+
+function getPharmacyResults(searchTerm) {
+  const uriSafeSearch = searchTerm.toLowerCase().replace(' ', '+')
+  const newLocal = "oscarRx/managePharmacy.do?method=search&search&term=" + uriSafeSearch;
+  var url =
+    getOrigin() +
+    "/" +
+    getProvider() +
+    "/" +
+    newLocal
+
+  return fetch(url, {
+    method: "GET",
+    headers: {
+      Accept: "text/javascript, text/html, application/xml, text/xml, */*",
+      "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+      "Content-Type": "application/json",
+    },
+  })
+}
+
+function getCurrentPharmacy(demographicNo) {
+  const newLocal = "oscarRx/managePharmacy.do?method=getPharmacyFromDemographic&demographicNo=" + demographicNo;
+  var url =
+    getOrigin() +
+    "/" +
+    getProvider() +
+    "/" +
+    newLocal
+
+  return fetch(url, {
+    method: "GET",
+    headers: {
+      Accept: "text/javascript, text/html, application/xml, text/xml, */*",
+      "Content-Type": "application/x-www-form-urlencoded; charset=UTF-8",
+      "Content-Type": "application/json",
+    },
+  })
+}
+
+function setPreferredPharmacy(pharmacyObj, demographicNo) {
+
+    const url =
+          getOrigin() +
+          "/" +
+          getProvider() +
+          "/" +
+          "oscarRx/managePharmacy.do?method=setPreferred"
+
+    pharmacyObj.name = pharmacyObj.name.replace(' ', '+')
+
+    var formData = new FormData();
+    formData.append("pharmacyId", String(pharmacyObj.id))
+    formData.append("demographicNo", String(demographicNo))
+    formData.append("pharmacyName", String(pharmacyObj.name))
+    formData.append("pharmacyAddress", String(pharmacyObj.address))
+    formData.append("pharmacyCity", String(pharmacyObj.city))
+    formData.append("pharmacyProvince", String(pharmacyObj.province))
+    formData.append("pharmacyPostalCode", String(pharmacyObj.postalCode))
+    formData.append("pharmacyPhone1", String(pharmacyObj.phone1))
+    formData.append("pharmacyPhone2", String(pharmacyObj.phone2))
+    formData.append("pharmacyFax", String(pharmacyObj.fax))
+    formData.append("pharmacyEmail", String(pharmacyObj.email))
+    formData.append("pharmacyServiceLocationId", String(pharmacyObj.serviceLocationIdentifier))
+    formData.append("pharmacyNotes", String(pharmacyObj.notes))
+    formData.append("preferredOrder", "1")
```

Comment:
Anything you append to formData automatically gets converted to string, so the extra work here may not be necessary (I think)